langlib/langlib-test/src/test/resources/test-src/xmllib_constrained_negative_test.bal

```diff
@@ -0,0 +1,42 @@
+// Copyright (c) 2020 WSO2 Inc. (http://www.wso2.org) All Rights Reserved.
+//
+// WSO2 Inc. licenses this file to you under the Apache License,
+// Version 2.0 (the "License"); you may not use this file except
+// in compliance with the License.
+// You may obtain a copy of the License at
+//
+// http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing,
+// software distributed under the License is distributed on an
+// "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+// KIND, either express or implied.  See the License for the
+// specific language governing permissions and limitations
+// under the License.
+
+import ballerina/lang.'xml;
+
+public function invalidSubtype() {
+    'xml:Comment c1 = xml `<hello>xml content</hello>`;
+    'xml:ProcessingInstruction p1 = xml `<hello>another xml content</hello>`;
+}
+
+public function invalidConstraint() {
+    xml<'xml:Comment> c1 = xml `<hello>xml content</hello>`;
+    xml<'xml:ProcessingInstruction> p1 = xml `<hello>another xml content</hello>`;
+
+    xml<'xml:Element> elementSequence = xml `<hello>xml content</hello>`;
+    'xml:Comment refer = elementSequence[0];
+
+    'xml:Comment comment = xml `<!-- this is a comment text -->`;
+    xml<'xml:Element> commentSequence = comment;
+}
+
+public function xmlConstraintMultipleElement() {
+    'xml:Element elem1 = xml `<hello>xml element 1</hello>`;
+    'xml:Element elem2 = xml `<hello>xml element 2</hello>`;
+    xml<'xml:Comment> elementSequence = elem1 + elem2;
+
+    'xml:Comment comment = xml `<!-- this is a comment text -->`;
+    xml<'xml:Element> seq = elem1 + comment;
+}
```

Comment:
Shall we add few negative test cases for invalid assignment to an array of xml subtype and invalid assignment to map of xml subtype tests?

xml<Comment>[] ar = [xml `<elem></elem>`];
ar.push(xml `<!-- cmnt -->`);
ar.push(xml `<?Hello DATA ?>`);

Reply:
Added via https://github.com/ballerina-platform/ballerina-lang/pull/22323/commits/fdec8352d3fd3b30ce2f41d75a5bbdc3ee1d5e3a